### PR TITLE
Add change signer method

### PR DESCRIPTION
**Pull request by fdarian**

### CODE REVIEW

File: src/SocioCatClaim.sol [hunk 1]
```diff
@@ -14,11 +15,17 @@ contract SocioCatClaim {
 
     error InvalidSignature();
     error InvalidNonce();
+    error ZeroAddress();
 
     event Claimed(address indexed to, uint256 amount);
+    event SignerSet(address indexed signer);
```
Comment (pyk) [inline src/SocioCatClaim.sol]: /SignerSet/SignerUpdated

**Review by pyk — CHANGES_REQUESTED** [review]

File: test/SocioCatClaim.t.sol [hunk 2]
```diff
@@ -117,6 +120,46 @@ contract SocioCatClaimTest is Test {
...
+        vm.startPrank(owner);
+        claim.setSigner(address(0));
+        vm.stopPrank();
+    }
+
+    function test_setSigner_rejectsNotOwner() public {
+        (address newSigner, ) = makeAddrAndKey("newSigner");
+        // --
+
+        vm.expectRevert();
```
Comment (pyk) [inline test/SocioCatClaim.t.sol]: specify revert selector here, make sure it revert `OwnableUnauthorizedAccount`

**Review by pyk — CHANGES_REQUESTED** [review]
LGTM 

just minor testing change

File: test/SocioCatClaim.t.sol (hunk 2 above)
Comment (fdarian) [reply test/SocioCatClaim.t.sol]: Done [88b4e4b](https://github.com/risedle/gatcha-contracts/pull/3/commits/88b4e4bde5a16809a9969c29cc036dc5c60f28cc)

**Review by pyk — APPROVED** [review]

**Pull request merged**